src/Data/Text/Internal.hs

```diff
@@ -85,8 +85,8 @@ text_ arr off len =
 
 -- | /O(1)/ The empty 'Text'.
 empty :: Text
-empty = Text A.empty 0 0
-{-# INLINE [1] empty #-}
+empty = empty_
+{-# INLINE empty #-}
```

Comment:
Do we still need `empty_`?

Reply:
No I just kept it in order to avoid removing exports. But considering this is an internal module I guess that's not a problem. Say the word and I'll remove it.

Reply:
Yeah I think you can remove it. I think originally they thought it might be useful to have both inlineable and non-inlineable variants, but this issue boils down to us not needing the inlineable one. All text operations are going to be looking at the length first, so a rewrite rule will take care of inlining `empty`.

Reply:
Sure makes sense. Done in ad91a4440bcfc934ad4c26523bdee20ec45b8d8